e2e/ctl_v3_alarm_test.go

```diff
@@ -15,44 +15,62 @@
 package e2e
 
 import (
+	"os"
 	"strings"
 	"testing"
+	"time"
 )
 
-func TestCtlV3Alarm(t *testing.T) { testCtl(t, alarmTest, withQuota(64*1024)) }
+func TestCtlV3Alarm(t *testing.T) {
+	// The boltdb minimum working set is six pages.
+	testCtl(t, alarmTest, withQuota(int64(48*os.Getpagesize())))
+}
 
 func alarmTest(cx ctlCtx) {
 	// test small put still works
-	smallbuf := strings.Repeat("a", int(cx.quotaBackendBytes/100))
-	if err := ctlV3Put(cx, "abc", smallbuf, ""); err != nil {
+	smallbuf := strings.Repeat("a", 64)
+	if err := ctlV3Put(cx, "1st_test", smallbuf, ""); err != nil {
 		cx.t.Fatal(err)
 	}
 
-	// test big put (to be rejected, and trigger quota alarm)
-	bigbuf := strings.Repeat("a", int(cx.quotaBackendBytes))
-	if err := ctlV3Put(cx, "abc", bigbuf, ""); err != nil {
-		if !strings.Contains(err.Error(), "etcdserver: mvcc: database space exceeded") {
-			cx.t.Fatal(err)
+	// write some small chunks to fill up the database
+	buf := strings.Repeat("b", int(os.Getpagesize()/3))
```

Comment:
When we run on a system with 64KiB pages, we can't write a buffer large enough to hit the quota limit because we hit the command line limit of etcdctl (128 KiB).  That's why I changed this to fill up the DB with smaller puts until we do hit the quota limit.  What do you suggest?